### PR TITLE
WIP: Balefinder improvements and #2816

### PR DESCRIPTION
**Pull request by schwiti6190**
- Implemented bale manager with lock and unlock features for bales
- Changed bale finder pathfinding to new controller

### CODE REVIEW

File: scripts/ai/BaleToCollect.lua [hunk 1]
```diff
@@ -136,6 +136,59 @@ function BaleToCollect:getSafeDistance()
...
+
+function BaleToCollectManager:init()
+	self.temporaryLockedBales = {}
+	self.lockedBales = {}
+end
+
+function BaleToCollectManager:update(dt)
+	for bale, time in pairs(self.temporaryLockedBales) do 
+		if time < (g_time + self.lockTimeOutMs) then 
+			self.temporaryLockedBales[bale] = nil
```
Comment (pvaiko) [inline scripts/ai/BaleToCollect.lua:151]: I would have done this with a CpTemporaryObject so there is no need for a cyclic update.

File: scripts/ai/AIDriveStrategyFindBales.lua [hunk 2]
```diff
@@ -547,10 +560,25 @@ function AIDriveStrategyFindBales:update(dt)
...
     if self:areBaleLoadersFull() and self:isReadyToFoldImplements() then
         self:debug('Bale loader is full, stopping job.')
         self.vehicle:stopCurrentAIJob(AIMessageErrorIsFull.new())
     end
+
+    --- Ignores the loaded auto loader bales.
+    --- TODO: Maybe add a delay?
+    local loadedBales = self:getBalesToIgnore()
+    for _, bale in pairs(loadedBales) do 
+        g_baleToCollectManager:lockBaleTemporary(bale)
```
Comment (pvaiko) [inline scripts/ai/AIDriveStrategyFindBales.lua:580]: Why is this needed? Isn't that bale already locked when we started pathfinding? A few comments to explain when we lock/unlock and why would be useful.

**Review by Tensuko — APPROVED** [review]
